### PR TITLE
Origin/sbw imporving bpe

### PR DESCRIPTION
**Pull request by Daviderikmollberg**
It's not as a big PR as it loooks like!

- Removing the useless "all" dir
- Changed folders that were called "training" to "train"
- Changed directories which were "subword" to "sw" to shorten the dir
- Made runV3.1.sh, ignore runV3.sh. I will change it to runV3, I'm keeping versions for quick, but I see now that it makes this PR look scary and will stop that. I changed stage 2 and the beginning of stage 3 to handle multiple subword methods, with one parameter. I only have BPE implemented atm.
- Added the files apply_bpe.py and learn_bpe.py to sw_methods/bpe/ the files have not been modified at all.

### CODE REVIEW

File: s5_subwords/local/sw_methods/bpe/prepare_subword_text.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# 2019 Dongji Gao
```
Comment (judyfong) [inline s5_subwords/local/sw_methods/bpe/prepare_subword_text.sh:3]: What license is this file released under? Apache 2.0? MIT?

File: s5_subwords/runV3.1.sh [hunk 2]
```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 QCRI (author: Ahmed Ali)
+#           2019 Dongji Gao
```
Comment (judyfong) [inline s5_subwords/runV3.1.sh:4]: Your name should likely be here

File: s5_subwords/runV3.sh [hunk 3]
```diff
@@ -45,7 +45,8 @@ method='bpe'
 AUDIO=/data/asr/malromur/malromur2017/correct
 
 #samromur_root=/data/asr/samromur/samromur_ldc
-METADATA=/data/asr/malromur/malromur2017/malromur_metadata.tsv
+# METADATA=/data/asr/malromur/malromur2017/malromur_metadata.tsv
+METADATA=/home/derik/work/tools/normalize/malromur/normalized_files/malromur_metadata_subset.tsv
```
Comment (judyfong) [inline s5_subwords/runV3.sh:49]: If this is static, you should add it to the preprocessed directory like the samromur and samromur teen metadata files.

**Review by judyfong — COMMENTED** [review]
I only did a surface level review so I'll leave it to inga to approve and merge this PR

File: s5_subwords/runV3.sh (hunk 3 above)
Comment (Daviderikmollberg) [reply s5_subwords/runV3.sh:49]: I'm not sure I'm following. This file is only a temp file to speed up dev

File: s5_subwords/local/sw_methods/bpe/prepare_subword_text.sh (hunk 1 above)
Comment (Daviderikmollberg) [reply s5_subwords/local/sw_methods/bpe/prepare_subword_text.sh:3]: Apache 2.0 I would guess as it is stated in the run file they created orignally

Comment (judyfong) [reply s5_subwords/local/sw_methods/bpe/prepare_subword_text.sh:3]: ok can you add that to the file?

File: s5_subwords/runV3.sh (hunk 3 above)
Comment (judyfong) [reply s5_subwords/runV3.sh:49]: ok if it's only temp, no need to do anything

File: s5_subwords/runV3.1.sh (hunk 2 above)
Comment (Daviderikmollberg) [reply s5_subwords/runV3.1.sh:4]: Noted

**Comment by Daviderikmollberg** [comment]
All these changes are now in the branch unigram_update

**Pull request closed without merging**